### PR TITLE
feature/7814 UpdateProfileScreensButtonsAndLinks

### PR DESCRIPTION
**Pull request by theodur**
## Description of Change

Updates buttons and links on Profile related screens to be more accessible. Most notable changes are:
- Updates the default accessibility role of a base list item to `menuitem` instead of `button`
- Adds an alert when the `Reset first time login` button is tapped to confirm actions
- Displays a Snackbar when after the `Reset in-app review actions` button is tapped and the review actions have been successfully reset
- Converts the `Firebase debug mode` button to a toggle
- Turns the `Remote Config` button into a link/menuitem.
- Adds an alert when opening the system settings in the Push notification settings screen

**Note**: Keeping this PR in draft until I get confirmation from Brea whether some of the changes I made are okay.

## Screenshots/Video

<details><summary>Push notification system settings alert</summary>
</details>

<details><summary>Updated Developer screen buttons</summary>
</details>

## Testing

Tested locally in demo mode.
- [ ] Tested on iOS 
- [x] Tested on Android 

## Reviewer Validations

- [ ] **Addresses + Phone numbers + Email address**
     - **Location:** Profile / Contact information
     - **Current component type:** Button
     - **Current style:** N/A
     - **Icon:** Chevron Right
     - **What happens?** Launch subtask screens with options to add, update, and/or remove contact information
     - **Recommended change:** Should be links (or menu items)
- [ ] **Manage account**
     - **Location:** Profile / Settings
     - **Current component type:** Button
     - **Current style:** N/A
     - **Icon:** Chevron Right
     - **What happens?** Redirects to the manage account child screen
     - **Recommended change:** Should be a link (or a menu item)
- [ ] **Privacy policy**
     - **Location:** Profile / Settings
     - **Current component type:** Button
     - **Current style:** N/A
     - **Icon:** Chevron Right
     - **What happens?** Launches a native alert that states that the user is leaving the app 
     - **Recommended change:** Should be a link (or a menu item)
- [ ] **Developer screen**
     - **Location:** Profile / Settings
     - **Current component type:** Button
     - **Current style:** N/A
     - **Icon:** Chevron Right
     - **What happens?** Redirects to the developer screen
     - **Recommended change:** Should be a link (or a menu item)
- [ ] **Reset first time login**
     - **Location:** Profile / Settings / Developer screen
     - **Current component type:** Button
     - **Current style:** Primary
     - **Icon:** N/A
     - **What happens?** Resets the app to the first time login state and takes you back to the login screen
     - **Recommended change:** I would add an "are you sure?" messaging rather than just redirecting back to the login screen. There is the potential of an accidental tap and for something of this magnitude, I would at least have some type of confirmation / native alert messaging.
- [ ] **Reset in-app review actions**
     - **Location:** Profile / Settings / Developer screen
     - **Current component type:** Button
     - **Current style:** Primary
     - **Icon:** N/A
     - **What happens?** Resets in-app review actions
     - **Recommended change:** There is no verification / notification / validation that this button does anything.
- [ ] **Enable Firebase debug mode**
     - **Location:** Profile / Settings / Developer screen
     - **Current component type:** Button
     - **Current style:** Primary
     - **Icon:** N/A
     - **What happens?** Toggles / words change from "enable..." to "disable..." / Turns debug mode on and off
     - **Recommended change:** This should be a toggle not a button
- [ ] **Remote config**
     - **Location:** Profile / Settings / Developer screen
     - **Current component type:** Button
     - **Current style:** Primary
     - **Icon:** N/A
     - **What happens?** Redirects to the remote config child screen
     - **Recommended change:** Should be a link and not a button
- [x] **Notifications**
     - **Location:** Profile / Settings / Notifications
     - **Current component type:** Button
     - **Current style:** N/A
     - **Icon:** Chevron Right
     - **What happens?** Redirects to the notifications child screen
     - **Recommended change:** Should be a link (or a menu item)
- [ ] **Open settings**
     - **Location:** Profile / Settings / Notifications
     - **Current component type:** Button
     - **Current style:** Primary
     - **Icon:** N/A
     - **What happens?** Opens the device's settings
     - **Recommended change:** Should alert a user that they will be leaving the app and launching their device's settings.

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [ ] PR is connected to issue(s)
- [ ] Tests are included to cover this change (when possible)
- [ ] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [ ] No secrets or API keys are checked in
- [ ] All imports are absolute (no relative imports)
- [ ] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

**Comment by theodur** [comment]
@timwright12 Yeah this is ready for review. I nudged global for reviews on my outstanding PRs last week. A few of them were approved, but this one wasn't yet. Binny is out, but I'll follow up with Chika again to see if she can get to this

File: VAMobile/src/components/BaseListItem.tsx [hunk 1]
```diff
@@ -228,7 +228,7 @@ const BaseListItem: FC<BaseListItemProps> = (props) => {
   }
 
   // Default role for list item is button
```
Comment (cadibemma) [inline VAMobile/src/components/BaseListItem.tsx:230]:
```suggestion
  // Default role for list item is menuitem
```

**Review by cadibemma — APPROVED** [review]
Looks good! Just one extra minor suggestion

**Review by cadibemma — APPROVED** [review]

**Comment by TKDickson** [comment]
Disability ratings page had some screenreader issues; see comment on ticket

**Review by TKDickson — APPROVED** [review]
OK I think the RC recut has been taken, so adding my official signoff for this to merge to develop

**Pull request merged**